network/interface.go

```diff
@@ -124,4 +124,7 @@ type Network interface {
 	NumConnectedPeers() int
 	ReachabilityStatus() string
 	HostAddrs() []string
+	Name() string
+	DHTSize() int32
+	Protocols() []string
```

Comment:
Supported protocols can get from other peers.

Reply:
Do you mean the `GetPeersInfo` method?

Reply:
Protocol negotiation. I don't think LibP2P returns all supported protocols by a node.

Reply:
Protocol negotiation works on Pactus network, this function is used for network info in gRPC and HTTP modules.

Reply:
How do you want to extract the protocol? The current implementation is wrong. It just return stream protocol id two times: `/pactus-testnet-v2/stream/v1, /pactus-testnet-v2/stream/v1`

The supported protocols should get from LibP2p2, like:
`/floodsub/1.0.0,/ipfs/id/1.0.0,/ipfs/id/push/1.0.0,/libp2p/circuit/relay/0.2.0/stop,/libp2p/dcutr,/meshsub/1.0.0,/meshsub/1.1.0,/pactus-testnet-v2/stream/v1`

Reply:
yeap, current approach is not fine. let me update that.

Reply:
fixed.